KeyVault.Acmebot/Providers/GoDaddyProvider.cs

```diff
@@ -0,0 +1,207 @@
+            {
+                var response = await _httpClient.GetAsync($"v1/domains/{zoneId}/records");
+
+                response.EnsureSuccessStatusCode();
+
+                var entries = await response.Content.ReadAsAsync<List<DnsEntry>>();
+
+                return entries;
+            }
+
+            public async Task DeleteRecordAsync(string zoneId, DnsEntry entry)
+            {
+                var response = await _httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Delete, $"v1/domains/{zoneId}/records/{entry.Type}/{entry.Name}"));
+
+                response.EnsureSuccessStatusCode();
+            }
+
+            public async Task AddRecordAsync(string zoneId, List<DnsEntry> entries)
+            {
+                HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Patch, $"v1/domains/{zoneId}/records")
+                {
+                    Content = new StringContent(JsonConvert.SerializeObject(entries, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore }), Encoding.UTF8, "application/json")
+                };
+
+                var response = await _httpClient.SendAsync(request);
+
+                response.EnsureSuccessStatusCode();
+            }
+
+            private sealed class ApiKeyHandler : DelegatingHandler
+            {
+                private string ApiKey { get; }
+                private string ApiSecret { get; }
+
+                public ApiKeyHandler(string apiKey, string apiSecret, HttpMessageHandler innerHandler) : base(innerHandler)
+                {
+                    if (apiKey is null)
+                    {
+                        throw new ArgumentNullException(nameof(apiKey));
+                    }
+
+                    if (apiSecret is null)
+                    {
+                        throw new ArgumentNullException(nameof(apiSecret));
+                    }
+
+                    if (innerHandler is null)
+                    {
+                        throw new ArgumentNullException(nameof(innerHandler));
+                    }
+
+                    if (string.IsNullOrWhiteSpace(apiKey))
+                    {
+                        throw new ArgumentException("API Key must be specified", nameof(apiKey));
+                    }
+
+                    if (string.IsNullOrWhiteSpace(apiSecret))
+                    {
+                        throw new ArgumentException("API Secret must be specified", nameof(apiSecret));
+                    }
+
+                    ApiKey = apiKey;
+                    ApiSecret = apiSecret;
+                }
+
+                protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+                {
+                    string authorization = $"sso-key {ApiKey}:{ApiSecret}";
+
+                    request.Headers.Add("Authorization", authorization);
+
+                    return base.SendAsync(request, cancellationToken);
```

Comment:
`ApiKey` / `ApiSecret` do not change during operation, so adding them to `DefaultRequestHeaders` should not be a problem.

Reply:
Agreed

Reply:
This fix is necessary; ApiKeyHandler is no longer needed.